contributing/core/developing-using-local-app.md

```diff
@@ -0,0 +1,70 @@
+# Developing Using Your Local Version of Next.js
+
+There are two options to develop with your local version of the codebase:
+
+### Set as a local dependency in package.json
+
+1. In your app's `package.json`, replace:
+
+   ```json
+   "next": "<next-version>",
+   ```
+
+   with:
+
+   ```json
+   "next": "file:/path/to/next.js/packages/next",
+   ```
+
+2. In your app's root directory, make sure to remove `next` from `node_modules` with:
+
+   ```sh
+   rm -rf ./node_modules/next
+   ```
+
+3. In your app's root directory, run:
+
+   ```sh
+   pnpm i
+   ```
+
+   to re-install all of the dependencies.
+
+   Note that Next will be copied from the locally compiled version as opposed to from being downloaded from the NPM registry.
+
+4. Run your application as you normally would.
+
+5. To update your app's dependencies, after you've made changes to your local `next` repository. In your app's root directory, run:
+
+   ```sh
+   pnpm install --force
+   ```
```

Comment:
I am using `pnpm@7.9.3`, and I personally don't have to re-run `pnpm i --force` every time I make a change.

I just keep `pnpm run dev` in `next` (watching file changes and re-build), and the app's `node_modules/next` will "automatically get updated". It seems that the newer version of pnpm will create a hard link.

Reply:
Updated: the newer version of pnpm does "link" folders into `node_modules`. So `pnpm install --force` is not required anymore.

Reply:
Continuing the discussion here: https://github.com/vercel/next.js/pull/40515#discussion_r969623609